### PR TITLE
Fix requirements/adjust_versions.py

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Noticed it was failing due to an outdated version mapping

Failure: https://github.com/PyTorchLightning/pytorch-lightning/runs/2404985974

Another:

```console
#7 23.44 The conflict is caused by:
#7 23.44     The user requested torch==1.8.1
#7 23.44     torchtext 0.9.0 depends on torch==1.8.0
#7 23.44 
#7 23.44 To fix this you could try to:
#7 23.44 1. loosen the range of package versions you've specified
#7 23.44 2. remove package versions to allow pip attempt to solve the dependency conflict
```

## Before submitting
- [n/a] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes? (if necessary)
- [n/a] Did you write any new necessary tests? (not for typos and docs)
- [n/a] Did you verify new and existing tests pass locally with your changes?
- [n/a] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Review by Borda — APPROVED** [review]
lgtm

**Comment by carmocca** [comment]
The version issue is now fixed, but it still fails because a package is uninstalling torch nightly and installing 1.8.1

Marking as draft until resolved

File: requirements/adjust_versions.py [hunk 1]
```diff
@@ -1,47 +1,50 @@
...
-    "1.7.1": dict(torchvision="0.8.2", torchtext="0.8.1"),
-    "1.8.0": dict(torchvision="0.9.0", torchtext="0.9"),
-    "1.8.1": dict(torchvision="0.9.0", torchtext="0.9"),
-}
-
-
-def find_latest(ver: str, versions_all: list) -> str:
+from typing import Dict, Optional
+
+# IMPORTANT: this list must be sorted from latest to oldest
```
Comment (tchaton) [inline requirements/adjust_versions.py:6]: Could you sort it using the torch key ?

File: requirements/adjust_versions.py [hunk 2]
```diff
@@ -1,47 +1,50 @@
...
+    with open(path_req, "r") as fp:
+        req = fp.read()
+
+    latest = find_latest(torch_version)
+    for lib, version in latest.items():
+        replace = f"{lib}=={version}" if version else lib
+        replace += os.linesep
         req = re.sub(rf"{lib}[>=]*[\d\.]*{os.linesep}", replace, req)
 
+    print(req)
```
Comment (tchaton) [inline requirements/adjust_versions.py:47]: Do we need to print ?

Comment (carmocca) [reply requirements/adjust_versions.py:47]: It's helpful as this is used in CI and the print shows the changes to the file.

But we don't *need* it

**Review by awaelchli — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Review by edgarriba — APPROVED** [review]

**Pull request merged**